rust/lance-core/src/datatypes.rs

```diff
@@ -202,6 +201,7 @@ impl TryFrom<&LogicalType> for DataType {
             "binary" => Some(Binary),
             "large_string" => Some(LargeUtf8),
             "large_binary" => Some(LargeBinary),
+            "json" => Some(LargeBinary),
```

Comment:
Does this mean JSON is always large binary?

Reply:
Yes for now. I'm open to provide `Binary` support for small json. Maybe it's worth a seperate issue?